vncviewer/vncviewer.cxx

```diff
@@ -503,6 +503,30 @@ static int mktunnel()
 }
 #endif /* !WIN32 */
 
+static int mainloop(const char* vncserver, network::Socket* sock)
+{
+  if(exitError != NULL) {
+    free(exitError);
+    exitError = NULL;
+  }
+
+  CConn *cc = new CConn(vncServerName, sock);
+
+  inMainloop = true;
+  exitMainloop = false;
+  while (!exitMainloop)
+    run_mainloop();
+  inMainloop = false;
+
+  delete cc;
+
+  if (exitError != NULL && alertOnFatalError)
+    return fl_choice("%s. Retry?", "No", "Yes", 0, exitError);
```

Comment:
As far as I understand the code we reach this line only in case before a successful connection was done. In case a connection could not be established the code errors out in `exit_vncviewer` around line 122. This is at least what I observed in my tests.

Reply:
Oh, so this PR doesn't handle a connection drop while working, so to speak? That makes the title of the PR quite misleading

Reply:
No, sorry, the other way round. This PR only handles the case when the connection before was successful. But not the case when no connection can be established initially.

Reply:
Ah, okay. In that case, wouldn't "Reconnect?" be a better question compared to "Retry?"?

Reply:
Yes, will change it.